website_event_compassion/models/account_invoice.py

```diff
@@ -53,3 +61,9 @@ def action_invoice_paid(self):
         task = self.env.ref("website_event_compassion.task_full_payment")
         registrations.write({"completed_task_ids": [(4, task.id)]})
         return res
+
+    @api.multi
+    def auto_cancel(self):
+        invoices = self.filtered(lambda i: i.state in ["open"])
```

Comment:
Maybe we should as well consider cancelling draft invoices?

```suggestion
        invoices = self.filtered(lambda i: i.state in ["draft", "open"])
```